Status report of background services for software

dmijatovic opened the pull request:
# Status report of background services

**To test background services you will need to have properly defined service items in the .env file. Some might require secret to enable service to scrape information using "less restricted" quota.**

Changes proposed in this pull request:
* The software maintainer can view status of the background services running for the software item
* The statuses reported are: 
    *  `scheduled`: if the url is provided and last_error or last_run value is not present
    * `not_active`: if the repository url or package manager url value is not present
    * `error`: if the last_error value is present
    * `success`: if the last_run value is present and the last_error value is NOT present. We show date/time of the last successful run
* Short description about the background services is added in the right panel of the page

How to test:
* `make start` to build app and generate test data
* login as user, if you use dev environment variable first user will become admin.
* navigate to first software item in the overview and go to edit software page.
* initially you should see background report with scheduled status for all items as in the image 1
* start the scrapers services by restarting app: `docker compose down && docker compose up`
* navigate to second software item. Remove repository URL and all package managers. The background service reports should like ones in the image 2
* login as rsd_amin (if not already), navigate to error logs page of administration section. From there use link to open the software page with an service error. Confirm that error is shown in the background services too (image 3)
* from the software overview page select the item that has programming languages in the card. Navigate to edit section and confirm that, at least, programming languages service was runned successfully (image 4)
* read the background services info on the right side of the page. If not clear please provide suggested changes to the explanation

## Background services (scheduled - initial state)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/f675697b-ec6b-41ac-a475-f572f86140ea)

## Background services (not active state)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/7fb5dfd9-29db-4e1e-b071-53e3203aca6a)

## Background services (error state)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/ae55724f-3421-458a-aebb-6b4ccc26f0f2)

## Background services (multiple package managers)
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/3a2a182b-af35-4ee7-b562-88957b8cbcf8)


PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [ ] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

jmaassen commented:
I tested this PR and have some comments. Unsure if this originates in the scrapers of frontend. My setup is an RSD with a single software and the following info: 

Repo: 
https://github.com/GooglingTheCancerGenome/sv-callers
 
Package managers: 
https://anaconda.org/conda-forge/esmvaltool
https://pypi.org/project/ESMValTool/
https://hub.docker.com/r/esmvalgroup/esmvaltool
https://cran.r-project.org/web/packages/GGIR
https://mvnrepository.com/artifact/nl.junglecomputing/sbbi-upnplib
https://www.npmjs.com/package/aws-sdk

I see the many of the scrapers running successfully after leaving the RSD for some time. Some never seem to run however: 

![Screenshot 2023-12-19 at 16-59-41 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/b3bfc746-ad09-4caa-afe5-ca0227b5e55d)

As a user, it is difficult to figure out if the scrapers hasn't run because of some mistake I have made, or because of some external reason (waiting for my turn, scraper did too many requests, scraper only runs once a day, a missing scraper, etc.). Would there be a way to make this more clear?

Unrelated: I noticed that when making a mistake when cutting and pasting a package manager URL, the PM detection sometimes fails. For example: 

- add package manager
- paste "ttps://hub.docker.com/r/esmvalgroup/esmvaltool" (note the first "h" missing)
- PM detection says "other"
- fix by moving to start of line and typing "h"
- PM detection still says "other"
- move to end of line and type <space> 
- PM detection now works correctly

dmijatovic commented:
Thanks for the feedback @jmaassen. I have fixed repository stats flag (was not correct) and removed package manager services that are not implemented (yet). I made force push, please remove old local branch and pull again. 

![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/bc995ca5-593c-4094-84e5-66512635fb63)

jmaassen commented:
The feedback provided about the services is now much better. A few other comments: 

I would change the text in this text box: 

![Screenshot 2023-12-28 at 10-28-59 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/423464aa-2d42-4773-acad-b7765411f333)

to: 

```
No information about package managers is provided for this software
```

And this one: 

![Screenshot 2023-12-28 at 10-29-10 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/dc8a57ce-d28e-4efc-bc4c-d149b0c1ae08)

to: 

```
The RSD collects additional information about your software using the links you
have provided. For example, the software repository URL is used to collect 
information about the commit history and programming languages, and a 
package manager URL can be used to retrieve download numbers.  

This section provides feedback if the RSD could succesfully retrieve 
this information or if it has encountered an error. These may be cause by 
a URL that is missing or incorrect. 

Please contact us at rsd@esciencecenter.nl in case you need our 
assistance in solving any of the errors.
```

There is also a typo here: 

![Screenshot 2023-12-28 at 10-48-23 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/b5759c48-c806-4e12-8e45-3effb60401e7)

"Commit hisitory" should be "Commit history". I would also use the full "Repository statistics" instead of "Repository stats"

For these two: 

![Screenshot 2023-12-28 at 10-47-58 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/df499e8c-b7e5-418e-a319-a05db0e7a398)

![Screenshot 2023-12-28 at 10-48-13 Edit software Research Software Directory](https://github.com/research-software-directory/RSD-as-a-service/assets/1705533/781ac341-3894-4a72-b4cb-5d07b870de8e)

I think that the "public api" bit in the subtitle is not that useful for the average user? Maybe its better to say: 

```
Additional information extracted using the source code repository URL
```

and

```
Additional information extracted using package manager URLs 
```

dmijatovic commented:
@jmaassen Thanks for the feedback. I improved the messages and rebased this PR to the latest main (to include updates of  Next and Node).

ewan-escience reviewed (changes requested):
Works well in principle, I have three small suggestions:

- When a Git repository is from Bitbucket or `other`, the status page should report that therse are not supported (yet). They now stay scheduled forever.
- `Scheduled but not yet executed. Check the status again after 30 minutes.` I would change this into `Scheduled, check again in 30 minutes.`
- `Software repository` as a header, I would replace that with `Git repository`.
